resources/views/map/custom-js.blade.php

```diff
@@ -0,0 +1,182 @@
+<script type="text/javascript" src="{{ asset('js/vis.min.js') }}"></script>
+<script type="text/javascript">
+    function custommapLegendPctDefaultColour(pct) {
```

Comment:
This is almost enough functions to put them under their own namespace.  It is fine as is, just suggesting now as it is harder to change later.

Reply:
I've updated to use a namespace.  Does the revised code look good, or is the name too generic, or is there a different preferred way to initialise the namespace?

Reply:
Sorry, JS is definitely not my strong suit :)  Can't answer those questions. The name is good.